### PR TITLE
Good news of an important feature: time series implementation in Elasticsearch storage.

### PR DESCRIPTION
**Pull request by peng-yongsheng**
Because of the feature of Elasticsearch, it rebuilds the index after executing delete by query command.
That is a heavy operate, it will hang up the Elasticsearch server for a few seconds. However, there are above 500+ indexes. So, we create the index by day to avoid execute delete by query operate, then 
delete the index directly, this is a high performance operate, say goodbye to hung up.

### CODE REVIEW

File: oap-server/server-starter/src/main/assembly/application.yml [hunk 1]
```diff
@@ -52,6 +52,7 @@ core:
     - Day
     - Month
     # Set a timeout on metrics data. After the timeout has expired, the metrics data will automatically be deleted.
+    enableDataKeeperExecutor: ${SW_CORE_ENABLE_DATA_KEEPER_EXECUTOR:true} # Turn it off then automatically metrics data delete will be close.
```
Comment (wu-sheng) [inline oap-server/server-starter/src/main/assembly/application.yml]: FYI @hanahmily . This will be supported in new storage core.

**Comment by wu-sheng** [comment]
FYI @apache/skywalking-committers This is a very important change of ElasticSearch storage improvement. Please be advised and pay attention to this change.

**Comment by wu-sheng** [comment]
Also, I directly change this PR to
1. Set otherMetricsDataTTL default value as 0, if no value set in `application.yml`.
1. Polish documents. 

In [my comment](https://github.com/apache/skywalking/pull/2808#pullrequestreview-244629306), suggestion (1) is still todo. suggestion (2) could be in another PR, if you agree. @peng-yongsheng

**Comment by zkqcommon** [comment]
that's good，waited for a long time。

**Comment by zkqcommon** [comment]
that's good，waited for a long time。

**Comment by PrettyJerry** [comment]
good job , looking forwards to this feature a long time. thanks.

**Comment by peng-yongsheng** [comment]
> * Recommend rename `timeSeriesAble` to `capableOfTimeSeries`.
I agree and I will rename it in this PR.
> * We should consider to bind `deleteHistory` and `timeSeriesAble` with **processor**. Also we could remove `@ Storage` and create `builder` for `@Storage`.
I have also thought about this. The streaming scenes of SkyWalking are sperate by the processors and the `deleteHistory` and `timeSeriesAble` is the ability of the processors. We can set them in processors, not need to define in the storage.

**Comment by wu-sheng** [comment]
@peng-yongsheng Your test cases fail, please fix.

**Pull request merged**